client/lib/translator/index.js

```diff
@@ -0,0 +1,226 @@
+
+		if ( ! localeCode || ! this.languageJson ) {
+			debug( 'trying to initialize translator without loaded language' );
+			return;
+		}
+
+		if ( this.initialized ) {
+			return;
+		}
+
+		if ( ! userSettings.getSettings() ) {
+			debug( 'initialization failed because userSettings are not ready' );
+			return;
+		}
+
+		if ( ! this.isEnabled() ) {
+			debug( 'not initializing, not enabled' );
+			return;
+		}
+
+		debug( 'Successfully initialized' );
+		this.initialized = true;
+	};
+
+	/**
+	 * TODO: expand jsdoc comment
+	 * "Activated" means that the translator is toggled on, and wrapTranslate()
+	 *     will add the data tags that the translator needs.
+	 *
+	 * @returns {Boolean}
+	 */
+	isActivated = () => this.shouldWrapTranslations;
+
+	/**
+	 * TODO: expand jsdoc comment
+	 * "Enabled" means that the user has opted in on the settings page
+	 *     ( but it's false until userSettings has loaded)
+	 *
+	 * @returns {Boolean}
+	 */
+	isEnabled() {
+		const currentUser = user.get();
+
+		if ( ! currentUser || 'en' === currentUser.localeSlug || ! currentUser.localeSlug ) {
+			return false;
+		}
+
+		if ( ! userSettings.getSettings() ) {
+			return false;
+		}
+
+		if ( ! userSettings.getOriginalSetting( 'enable_translator' ) ) {
+			return false;
+		}
+
+		// restrict mobile devices from translator for now while we refine touch interactions
+		if ( isMobile() ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * TODO: expand jsdoc comment
+	 *
+	 * @returns {Boolean}
+	 */
+	toggle() {
+		if ( ! this.isActivated() ) {
+			// Wrap DOM elements and then activate the translator
+			this.shouldWrapTranslations = true;
+			debug( 'Translator activated' );
+		} else {
+			this.shouldWrapTranslations = false;
+			debug( 'Translator deactivated' );
+		}
+
+		i18n.reRenderTranslations();
+	}
+
+	/**
+	 * TODO: expand jsdoc comment
+	 *
+	 * @returns {String|Object}
+	 */
+	wrapTranslation( originalFromPage, displayedTranslationFromPage, optionsFromPage ) {
+		if ( ! this.isEnabled() || ! this.isActivated() ) {
+			return displayedTranslationFromPage;
+		}
+
+		if ( 'object' !== typeof optionsFromPage ) {
+			optionsFromPage = {};
+		}
+
+		if ( 'string' !== typeof originalFromPage ) {
+			debug( 'unknown original format' );
+			return displayedTranslationFromPage;
+		}
+
+		if ( optionsFromPage.translatable === false ) {
+			debug( `author defined ${ displayedTranslationFromPage } as not translatable.` );
+			return displayedTranslationFromPage;
+		}
+
+		const props = {
+			singular: originalFromPage,
+			locale: this.currentLocale,
+		};
+
+		// Has Context
+		if ( 'string' === typeof optionsFromPage.context ) {
+			props.context = optionsFromPage.context;
+		}
+
+		// Has Plural
+		if ( 'string' === typeof optionsFromPage.plural ) {
+			props.plural = optionsFromPage.plural;
+		}
+
+		// Has no translation in current locale
+		if ( isEmpty( this.languageJson[ originalFromPage ] ) ) {
+			props.untranslated = 'true';
```

Comment:
Just curious why you'd have the string `'true'` here.

Reply:
I'm assuming for the same reason there are a couple of string checks (`'string' === typeof...`) above. React treat this as a DOM element, and complains when there are non-standard attributes and non-string values.

https://reactjs.org/blog/2017/09/08/dom-attributes-in-react-16.html